Monotype/c2pa-rs
# C2PA-484: Replace existing DSIG with dummy DSIG after signing

## Changes in this pull request

This PR modifies font signing behavior for SFNT fonts (notably WOFF still has old behavior) such that any DSIG which comes in with a font to be signed will be replaced with a dummy DSIG.  It does this by:
- Adding in a new DSIG names table which allows us to determine if one existed on the incoming font
  - Notably the parser ignores any signatures on the table, as we'll be stomping them anyway
- Looking for an existing DSIG table when the C2PA table is added, and replacing it with a dummy DSIG
- Building up the new TDIR by hand
  - Previously we were sneakier about this, as we knew we'd only ever be adding or removing the whole C2PA table; now that we have two tables changing we can't just rely on an offset shift
  - This required that I add `len` and `checksum` calls to all tables, allowing us to generate TDIR entries for all tables

## Verification

- Check out [c2pa-rs](https://github.com/Monotype/c2pa-rs) with this branch
- Check out [c2patool](https://github.com/Monotype/c2patool)'s `feature/C2PA-262/localSfntSupport` branch in the same directory as `c2pa-rs`
- Build the `c2patool`
- Sign a font with an existing DSIG
- Verify the signed font validates with https://fonttools.monotype.com/validator/options/ (verifies that checksums are all correct and that TDIR is in the proper order)

## Checklist
- [x] This PR represents a single feature, fix, or change.
- [ ] All applicable changes have been documented.
- [ ] Any `TO DO` items (or similar) have been entered as GitHub issues and the link to that issue has been included in a comment.
